compiler/onnx-tools/onnx-dump.py

```diff
@@ -43,6 +43,28 @@ def _get_attribute_value(attr):
     return helper.get_attribute_value(attr)
 
 
+def _gather_value_infos(onnx_model):
+    vis = dict()
+
+    for mod_input in onnx_model.graph.input:
+        vis[mod_input.name] = mod_input.type
+
+    for mod_output in onnx_model.graph.output:
+        vis[mod_output.name] = mod_output.type
+
+    for vi in onnx_model.graph.value_info:
+        vis[vi.name] = vi.type
+
+    return vis
+
+
+def _type_format(type):
+    dtstr = _data_type_str(type.tensor_type.elem_type)
+    shape = type.tensor_type.shape
+    shape_ar = [-1 if dim.dim_value == 0 else dim.dim_value for dim in shape.dim]
```

Comment:
From https://onnx.ai/onnx/operators/onnx__Reshape.html

> At most one dimension of the new shape can be -1. In this case, the value is inferred from the size of the tensor and the remaining dimensions. A dimension could also be 0, in which case the actual dimension value is unchanged (i.e. taken from the input tensor).

It seems that dim_value 0 and -1 are different. May I ask why you changed 0 to -1?

Reply:
Ah.. it is only for Reshape Op. Please ignore this comment.